### PR TITLE
WIP - Env survey addition

### PR DESCRIPTION
**Pull request by squirrelo**
Adds the environmental sample survey and logging, so they have specific information attached to each sample.

### CODE REVIEW

File: amgut/db/patches/0034.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+----------------------------------------------------------
+-- surveys
+----------------------------------------------------------
+INSERT INTO surveys (survey_id, survey_group) VALUES (3, -3);
+
+----------------------------------------------------------
+-- survey_question
+----------------------------------------------------------
+INSERT INTO survey_question (survey_question_id, question_shortname, american, british) VALUES (162, 'ENV_NAME', 'Descriptive name for the sample', 'Descriptive name for the sample');
+INSERT INTO survey_question (survey_question_id, question_shortname, american, british) VALUES (163, 'ZIP', 'Zipcode sample was collected in', 'Zipcode sample was collected in');
```
Comment (wasade) [inline amgut/db/patches/0034.sql]: what about non-US collected samples, or samples in international waters?

File: amgut/handlers/env_survey.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        env_survey_id = self.get_argument('survey_id', None)
+        sitebase = media_locale['SITEBASE']
+        new = False
+        if not env_survey_id:
+            env_survey_id = binascii.hexlify(os.urandom(8))
+            new = True
+
+        form = self.env_survey()
+        form.process(data=self.request.arguments)
+        print form.data
```
Comment (wasade) [inline amgut/handlers/env_survey.py]: print statement

File: amgut/db/patches/0034.sql (hunk 1 above)
Comment (squirrelo) [inline amgut/db/patches/0034.sql]: Ah, should be postcode, but yeah, international waters is...interesting. Could also ask for lat/long directly and use that instead if given.

File: amgut/handlers/env_survey.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+        consent = {
+            'login_id': ag_login_id,
+            'participant_name': participant_name,
+            'participant_email': ag_login_info['email'],
+            'assent_obtainer': 'ENV_SURVEY',
+            'parent_1_name': 'ENV_SURVEY',
+            'parent_2_name': 'ENV_SURVEY',
+            'survey_id': env_survey_id,
+            'is_juvenile': True,
```
Comment (wasade) [inline amgut/handlers/env_survey.py]: ?

Comment (squirrelo) [inline amgut/handlers/env_survey.py]: We need a value there, and I'm assuming bacteria are under 18 years old. Same thing as happened with the animal survey.

File: amgut/lib/data_access/ag_data_access.py [hunk 4]
```diff
@@ -465,6 +461,16 @@ def getParticipantSamples(self, ag_login_id, participant_name):
...
             return [dict(row) for row in rows]
 
+    def getEnvSurveyNames(self, ag_login_id):
+        sql = """SELECT DISTINCT participant_name from ag.ag_login_surveys
+                 JOIN ag.survey_answers_other USING (survey_id)
+                 JOIN ag.group_questions gq USING (survey_question_id)
+                 JOIN ag.surveys ags USING (survey_group)
+                 WHERE ag_login_id = %s AND ags.survey_id = %s"""
+        with TRN:
+            TRN.add(sql, [ag_login_id, 3])
```
Comment (wasade) [inline amgut/lib/data_access/ag_data_access.py]: 3?

File: amgut/lib/data_access/ag_data_access.py [hunk 5]
```diff
@@ -465,6 +461,16 @@ def getParticipantSamples(self, ag_login_id, participant_name):
             rows = TRN.execute_fetchindex()
             return [dict(row) for row in rows]
 
+    def getEnvSurveyNames(self, ag_login_id):
```
Comment (wasade) [inline amgut/lib/data_access/ag_data_access.py]: it might be nice to have docstrings

File: amgut/lib/data_access/ag_data_access.py (hunk 4 above)
Comment (squirrelo) [inline amgut/lib/data_access/ag_data_access.py]: ENV survey is 3. Added comment.

File: amgut/lib/data_access/ag_data_access.py (hunk 5 above)
Comment (squirrelo) [inline amgut/lib/data_access/ag_data_access.py]: added

File: amgut/handlers/env_survey.py (hunk 2 above)
Comment (squirrelo) [inline amgut/handlers/env_survey.py]: removed

**Comment by squirrelo** [comment]
Closed in favor or RedCap integration work.

**Pull request closed without merging**